File: AppGenerateFiles/helpdesk/FrontModel/Cat_Dependencias.js
//@ts-check
import { EntityClass } from "../../WDevCore/WModules/EntityClass.js";
import { WAjaxTools } from "../../WDevCore/WModules/WComponentsTools.js";
//@ts-ignore
import { ModelProperty } from "../../WDevCore/WModules/CommonModel.js";
import { Cat_Dependencias }  from './Cat_Dependencias.js'
import { CaseTable_Agenda }  from './CaseTable_Agenda.js'
import { CaseTable_Calendario }  from './CaseTable_Calendario.js'
import { CaseTable_Case }  from './CaseTable_Case.js'
import { CaseTable_Dependencias_Usuarios }  from './CaseTable_Dependencias_Usuarios.js'
import { Cat_Dependencias }  from './Cat_Dependencias.js'
import { Tbl_Servicios }  from './Tbl_Servicios.js'
class Cat_Dependencias extends EntityClass {
   constructor(props) {
       super(props, 'EntityHelpdesk');
       for (const prop in props) {
           this[prop] = props[prop];
       }
   }
   /**@type {Number}*/ Id_Dependencia;
   /**@type {String}*/ Descripcion;
   /**@type {String}*/ Username;
   /**@type {String}*/ Password;
   /**@type {String}*/ Host;
   /**@type {String}*/ AutenticationType;
   /**@type {String}*/ TENAT;
   /**@type {String}*/ CLIENT;
   /**@type {String}*/ OBJECTID;
   /**@type {String}*/ CLIENT_SECRET;
   /**@type {String}*/ HostService;
   /**@type {String}*/ SMTPHOST;
   /**@type {Boolean}*/ Default;
   /**@type {Cat_Dependencias} ManyToOne*/ Cat_Dependencias;
   /**@type {Array<CaseTable_Agenda>} OneToMany*/ CaseTable_Agenda;
   /**@type {Array<CaseTable_Calendario>} OneToMany*/ CaseTable_Calendario;
   /**@type {Array<CaseTable_Case>} OneToMany*/ CaseTable_Case;
   /**@type {Array<CaseTable_Dependencias_Usuarios>} OneToMany*/ CaseTable_Dependencias_Usuarios;
   /**@type {Array<Cat_Dependencias>} OneToMany*/ Cat_Dependencias;
   /**@type {Array<Tbl_Servicios>} OneToMany*/ Tbl_Servicios;
}
export { Cat_Dependencias }

File: AppGenerateFiles/helpdesk/Model/Cat_Dependencias.cs
using CAPA_DATOS;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
namespace DataBaseModel {
   public class Cat_Dependencias : EntityClass {
       [PrimaryKey(Identity = true)]
       public int? Id_Dependencia { get; set; }
       public string? Descripcion { get; set; }
       public int? Id_Dependencia_Padre { get; set; }
       public string? Username { get; set; }
       public string? Password { get; set; }
       public string? Host { get; set; }
       public string? AutenticationType { get; set; }
       public string? TENAT { get; set; }
       public string? CLIENT { get; set; }
       public string? OBJECTID { get; set; }
       public string? CLIENT_SECRET { get; set; }
       public string? HostService { get; set; }
       public string? SMTPHOST { get; set; }
       public bool? Default { get; set; }
       [ManyToOne(TableName = "Cat_Dependencias", KeyColumn = "Id_Dependencia", ForeignKeyColumn = "Id_Dependencia_Padre")]
       public Cat_Dependencias? Cat_Dependencias { get; set; }
       [OneToMany(TableName = "CaseTable_Agenda", KeyColumn = "Id_Dependencia", ForeignKeyColumn = "Id_Dependencia")]
       public List<CaseTable_Agenda>? CaseTable_Agenda { get; set; }
       [OneToMany(TableName = "CaseTable_Calendario", KeyColumn = "Id_Dependencia", ForeignKeyColumn = "Id_Dependencia")]
       public List<CaseTable_Calendario>? CaseTable_Calendario { get; set; }
       [OneToMany(TableName = "CaseTable_Case", KeyColumn = "Id_Dependencia", ForeignKeyColumn = "Id_Dependencia")]
       public List<CaseTable_Case>? CaseTable_Case { get; set; }
       [OneToMany(TableName = "CaseTable_Dependencias_Usuarios", KeyColumn = "Id_Dependencia", ForeignKeyColumn = "Id_Dependencia")]
       public List<CaseTable_Dependencias_Usuarios>? CaseTable_Dependencias_Usuarios { get; set; }
       [OneToMany(TableName = "Cat_Dependencias", KeyColumn = "Id_Dependencia", ForeignKeyColumn = "Id_Dependencia_Padre")]
       public List<Cat_Dependencias>? Cat_Dependencias { get; set; }
       [OneToMany(TableName = "Tbl_Servicios", KeyColumn = "Id_Dependencia", ForeignKeyColumn = "Id_Dependencia")]
       public List<Tbl_Servicios>? Tbl_Servicios { get; set; }
   }
}
